mathjax3-ts/input/tex/color/ColorConfiguration.ts

```diff
@@ -0,0 +1,336 @@
+        if (!rgbParts[i].match(/^\d+$/))
+        {throw new TexError('InvalidNumber', 'Invalid number')}
+        var n = parseInt(rgbParts[i]);
+        if (n > 255) {
+            throw new TexError('ModelArg2',
+                'Color values for the %1 model must be between %2 and %3',
+                'RGB', '0', '255');
+        }
+        var pN = n.toString(16); 
+        if (pN.length < 2) {pN = '0'+pN}
+        RGB += pN;
+    }
+    return RGB;
+}
+
+/**
+ *  Get a gray-scale value
+ */
+function get_gray(gray: string) {
+    if (!gray.match(/^\s*(\d+(\.\d*)?|\.\d+)\s*$/)) {
+        throw new TexError('InvalidDecimalNumber','Invalid decimal number')
+    }
+
+    var n : number = parseFloat(gray);
+    if (n < 0 || n > 1) {    
+        throw new TexError('ModelArg2',
+            'Color values for the %1 model must be between %2 and %3',
+            'gray', "0", "1");
+    }
+    var pn = Math.floor(n*255).toString(16); 
+    if (pn.length < 2) {
+        pn = '0' + pn;
+    }
+
+    return '#' + pn + pn + pn;
+}
+
+/**
+ *  Get a named value
+ */
+function get_named(name: string) : string {
+    if (COLORS.has(name)) {
+        return COLORS.get(name);
+    } else {
+        return name;
+    }
+}
+
+
+export let ColorMethods: Record<string, ParseMethod> = {};
+
+
+/**
+ * Override \color macro definition
+*/
+ColorMethods.Color = function(parser: TexParser, name: string) {
+    const model = parser.GetBrackets(name, "");
+    const color = getColor(model, parser.GetArgument(name));
+    
+    const math = parser.ParseArg(name);
+    
+    const node = parser.configuration.nodeFactory.create('node', 'mstyle', [math], {mathcolor: color});
+    parser.stack.env['color'] = color;
+    parser.Push(node);
+}
+
+ColorMethods.TextColor = function(parser: TexParser, name: string) {
+    const model = parser.GetBrackets(name, "");
+    const color = getColor(model, parser.GetArgument(name));
+    const old = parser.stack.env['color'];
+    parser.stack.env['color'] = color;
+    const math = parser.ParseArg(name);
+    if (old) {
+      parser.stack.env['color'];  // TODO: Volker, shouldn't we remove this?
+    } else {
+      delete parser.stack.env['color'];
+    }
+    const node = parser.configuration.nodeFactory.create('node', 'mstyle', [math], {mathcolor: color});
+    parser.Push(node);
+}
+
+/**
+ * Define the \definecolor macro
+*/
+ColorMethods.DefineColor =  function(parser: TexParser, name: string) {
+    var cname = parser.GetArgument(name),
+        model = parser.GetArgument(name),
+        def = parser.GetArgument(name);
+    
+    COLORS.set(cname, getColor(model, def));
+};
+
+/**
+ * Produce a text box with a colored background
+*/
+ColorMethods.ColorBox = function (parser: TexParser, name: string) {
+    var cname = parser.GetArgument(name),
+    math = ParseUtil.internalMath(parser, parser.GetArgument(name));
+
+    const node = parser.configuration.nodeFactory.create('node', 'mpadded', [math], {
```

Comment:
I think `math` is already an array of MathML nodes, not a single one, so you should not need `[math]` here, only `math`.

Reply:
Done.